src/context/campaignConfigsStore.tsx

```diff
@@ -107,27 +134,80 @@ export const CampaignConfigsStoreContextProvider: FunctionComponent = ({
     setAllConfigs({});
   }, []);
 
-  const [, setState] = useState();
   useEffect(() => {
-    const loadCampaignConfigFromStore = async (): Promise<void> => {
-      const campaignConfigsString = await AsyncStorage.getItem(
-        CAMPAIGN_CONFIGS_STORE_KEY
-      );
-      if (!campaignConfigsString) {
-        setHasLoadedFromStore(true);
-        return;
+    /**
+     * Migrates credentials from old config store to new config store. Clears all old storage locations.
+     * @param newStorageHasData whether most recent data has been found. If this is true, does not
+     * attempt any migration and just clears old storage locations
+     * @returns the updated value of {@link newStorageHasData}, i.e. true if any updated data was found
+     * in the older storage (if {@link newStorageHasData} was passed in as true, it will always return true)
+     */
+    const migrateOldConfigFromStore = async (
+      newStorageHasData: boolean
+    ): Promise<boolean> => {
+      let hasUpdatedData = newStorageHasData;
+      // check v1 storage
+      if (!hasUpdatedData) {
+        const configString = await AsyncStorage.getItem(
+          CAMPAIGN_CONFIGS_STORE_KEY
+        );
+        if (configString) {
+          try {
+            const configStringFromStore: CampaignConfigsMap = JSON.parse(
+              configString
+            );
+            setAllConfigs(configStringFromStore);
+            hasUpdatedData = true;
+          } catch (e) {
+            setState(() => {
+              throw new CampaignConfigError(e);
+            });
+          }
+        }
+      }
+      if (hasUpdatedData) {
+        await AsyncStorage.removeItem(CAMPAIGN_CONFIGS_STORE_KEY);
       }
+
+      return hasUpdatedData;
+    };
+
+    const loadCampaignConfigFromStore = async (): Promise<void> => {
+      let newStorageHasData = false;
       try {
-        const allCampaignConfigsFromStore: CampaignConfigsMap = JSON.parse(
-          campaignConfigsString
+        const campaignConfigsString = await readFromStoreInBuckets(
+          CAMPAIGN_CONFIGS_STORE_KEY
         );
-        setAllConfigs(allCampaignConfigsFromStore);
-        setHasLoadedFromStore(true);
+        if (campaignConfigsString !== null) {
+          const campaignConfigsFromStore = JSON.parse(campaignConfigsString);
+          setAllConfigs((prev) => ({
+            ...prev,
+            ...campaignConfigsFromStore,
+          }));
+          newStorageHasData = true;
+        }
       } catch (e) {
         setState(() => {
           throw new CampaignConfigError(e);
         });
       }
+      setHasLoadedFromPrimaryStore(true);
+      const migrated = await migrateOldConfigFromStore(newStorageHasData);
```

Comment:
then you can pass `await migrateOldConfigFromStore(!!campaignConfigsString);` or if you want to be cleaner, create the `newStorageHasData` variable just before this instead.

Reply:
would it potentially be a problem if `campaignConfigsString` is an empty string `""`? Because then it would be a falsy value.
Although by right if it is an empty string the JSON parse would throw an error already, but then the issue would be with the conditions being slightly different (line 195 is non-null and line 209 is truthy). Do you think this is an issue?

Or we can also do `await migrateOldConfigFromStore(campaignConfigsString !== null);`